sqlmesh/utils/jinja.py

```diff
@@ -267,14 +267,23 @@ def build_environment(self, **kwargs: t.Any) -> Environment:
         if self.root_package_name is not None:
             package_macros[self.root_package_name].update(root_macros)
 
+        env = environment()
+
+        builtin_globals = self._create_builtin_globals(kwargs)
+        merged_top_level_package_macros: t.Dict[str, AttributeDict] = {}
         for top_level_package_name in self.top_level_packages:
             root_macros.update(package_macros.get(top_level_package_name, {}))
+            merged_top_level_package_macros[top_level_package_name] = AttributeDict(
```

Comment:
update package_macros here instead of introducing another var to simplify the code.